server/controllers/tasks/lib/build_task_docs.coffee

```diff
@@ -2,49 +2,86 @@ __ = require('config').universalPath
+getAuthorWorksData = require './get_author_works_data'
 hasWorksLabelsOccurrence = __.require 'controllers', 'entities/lib/has_works_labels_occurrence'
+{ turnIntoRedirection } = __.require 'controllers', 'entities/lib/merge_entities'
 { prefixifyInv } = __.require 'controllers', 'entities/lib/prefix'
+{ _id:reconcilerUserId } = __.require('couch', 'hard_coded_documents').users.reconciler
 
 module.exports = (entity)->
   Promise.all [
     searchEntityDuplicatesSuggestions entity
     getAuthorWorksData entity._id
   ]
-  .spread (suggestionEntities, authorWorksData)->
-    relationScore = calculateRelationScore suggestionEntities
-    Promise.all suggestionEntities.map(create(authorWorksData, relationScore))
-
-create = (authorWorksData, relationScore)->
-  { labels, langs } = authorWorksData
-  return (suggestionEntity)->
-    suggestionUri = suggestionEntity.uri
-    _.type suggestionUri, 'string'
-    hasWorksLabelsOccurrence suggestionUri, labels, langs
-    .then (hasOccurence)->
+  .spread (suggestions, authorWorksData)->
+    unless suggestions.length > 0 then return []
+    Promise.all suggestions.map getOccurrences authorWorksData
+    .then _.compact
+    .then turnSuggestionIntoRedirection(suggestions, authorWorksData)
+    .then (occurrences)->
+      Promise.all filterSuggestions(occurrences, suggestions)
+      .then createTasksDocs(authorWorksData, occurrences)
+
+turnSuggestionIntoRedirection = (suggestions, authorWorksData)->
+  return (occurrences) ->
+    unless occurrences.length > 0 then return occurrences
+    { labels, authorId } = authorWorksData
+    matchedTitles = getMatchedTitles(occurrences)
+    # Todo : check every labels to turn entity into redirection
+    unless canBeRedirected suggestions, matchedTitles then return occurrences
+    # assume first occurrence is the right one to merge into
+    # since only one suggestion necessary to merge
+    # occurrences of first suggestion picked
+    turnIntoRedirection reconcilerUserId, authorId, occurrences[0].uri
+    return []
+
+getMatchedTitles = (occurrences)->
+    matchedTitles = occurrences.map (occ)-> _.pluck(occ.occurrences,'matchedTitles')
+    _.flattenDeep matchedTitles
+
+filterSuggestions = (occurrences, suggestions)->
+  # create a task for every suggestions
+  unless occurrences.length > 0 then return suggestions
+  # create tasks only for suggestions with occurrences
+  suggestions.filter (suggestion)->
+    occurrencesUris = _.pluck occurrences, 'uri'
+    suggestion.uri in occurrencesUris
+
+createTasksDocs = (authorWorksData, occurrences) ->
+  return (suggestions) ->
+    relationScore = calculateRelationScore suggestions
+    return suggestions.map create(authorWorksData, relationScore, occurrences)
+
+canBeRedirected = (suggestions, matchedTitles) ->
+  # several suggestions == has homonym => cannot be redirected
+  unless suggestions.length == 1 then return false
+  longTitles = matchedTitles.filter (title) -> title.length > 12
```

Comment:
shouldn't there be also a critiria in the number of words?
```suggestion
  longTitles = matchedTitles.filter (title) -> title.length > 12 and title.split(' ').length > 1
```

Reply:
not sure its necessary, a word of 12 chars is as radom as 2 words of 12 chars total, which is also as random as 6 words of 1é chars total

Reply:
I don't think this statement is true, but we can still try like that and see what kind and quantity of false positive it generates, and adapt accordingly